### ISSUE TITLE
Restrict Function method names to be only valid identifiers

### ISSUE THREAD

**Issue by frol** [near/nearcore#3386]
***Describe the bug***

Through Indexer for Explorer (/cc @khorolets) we observed that some people played with invalid transactions being constructed, namely ones that have method names with NULL byte in it. I suggest we only allow certain naming for method names as we display them in various places and it has no sense to display `\u0000` bytes.

Proposed valid identifier for exported method names: `<span class="error">[a-zA-Z0-9_]</span>+`.

***To Reproduce***

Submit a function call transaction with method name containinig \0 (NULL byte).

See `method_name` in [https://explorer.testnet.near.org/transactions/2pQGWtcm3aGsUgzyvfFeTeBhrEb8DkEwyfceeebNYqf5:](https://explorer.testnet.near.org/transactions/2pQGWtcm3aGsUgzyvfFeTeBhrEb8DkEwyfceeebNYqf5:)

```json
 {
 "actions": [
 {
 "FunctionCall": 

 { "args": "eyJ0b2tlbiI6ImdvbGQubmVhcnN3YXAudGVzdG5ldCIsInRva2Vuc19vdXQiOiIxMjAwMDAwMDAwMDAwMDAwMDAwMDAwMDAifQ==", "deposit": "0", "gas": 30000000000000, "method_name": "\u0000" } 

 }
 ],
 "hash": "2pQGWtcm3aGsUgzyvfFeTeBhrEb8DkEwyfceeebNYqf5",
 "signer_id": "send.testnet"
 }
 ```

***Expected behavior***

Discard FunctionCall actions with invalid method names.

***Proposed solution***

I would propose to introduce a self-validating type like it is planned for AccountId (#2074), but it is not feasible now since the network has those transactions with "invalid" method names now, so we cannot just return an error on deserialization since that will break local deserialization, right?

***Version (please complete the following information):***

*   nearcore commit/branch: any
*   mainnet/testnet/betanet/local

/cc @bowenwang1996 @evgenykuzyakov @nearmax @olonho

**Comment by bowenwang1996** [comment]
Just want to point out that this requires a protocol change, but I think it's a good idea.

**Comment by evgenykuzyakov** [comment]
For validity check we need to see, what's allowed by WebAssembly standard. I know the unicode is allowed, but I don't think anyone is actually using it.

**Comment by bowenwang1996** [comment]
@frol is this still an issue for you? Ideally I'd like to avoid doing protocol changes unless there is a strong reason for it.

**Comment by khorolets** [comment]
@bowenwang1996 yes https://github.com/near/near-indexer-for-explorer/issues/109 but I guess we can handle it from indexer for explorer side for database. However it would be great to implement it and remember to include it if protocol change happens :)
